app/javascript/channels/testRunChannel.ts

```diff
@@ -0,0 +1,29 @@
+import consumer from '../utils/action-cable-consumer'
+import { camelizeKeys } from 'humps'
+import { Submission } from '../components/student/Editor'
+import { TestRun } from '../components/student/editor/TestRunSummary'
+import { typecheck } from '../utils/typecheck'
+
+export class TestRunChannel {
+  subscription: ActionCable.Channel
+
+  constructor(submission: Submission, received: (testRun: TestRun) => void) {
+    this.subscription = consumer.subscriptions.create(
+      {
+        channel: 'Submission::TestRunsChannel',
+        submission_uuid: submission.uuid,
+      },
+      {
+        received: (response: any) => {
+          const formattedResponse = camelizeKeys(response)
+
+          received(typecheck<TestRun>(formattedResponse, 'testRun'))
```

Comment:
I was slightly confused by the duplicate use of `received`, once for the callback and one for the object key. It might be worth considering changing the parameter to a different name. Maybe something like `onReceived`?

Reply:
Cool. Sounds good!

Reply:
(This would break with linting: no-shadow, so changing now = good idea)